### PR TITLE
Add JSON config compatibility and possible options for CMake

### PR DESCRIPTION
**Pull request by SamHerts**
Added the library nanojsonc since it is c99 compatible and event driven, to match the ini.h functions.
Added a CMakeLists.txt to Configurator to show possible options for automated installation of ini.h, tinyexpr, and nanojsonc.

### CODE REVIEW

File: configurator/configurator.c [hunk 1]
```diff
@@ -583,53 +585,183 @@ int inih_config_handler(void* user,
...
+    int rc;
+
+    if( NULL == cfg )
+        return rc;
     
+    if( NULL == file )
+        return EINVAL;
+
+    // Determine the filetype based on extension
+    char* ext = strrchr(file, '.') + 1;
```
Comment (MichaelBrim) [inline configurator/configurator.c]: when filename does not include a '.' `strrchr()` may return NULL/0, leading to bad ext pointer value

File: configurator/configurator.c [hunk 2]
```diff
@@ -583,53 +585,183 @@ int inih_config_handler(void* user,
...
+
+int json_parse(const char* filename, NanoJSONCCallback json_config_handler, prefix_cfg_t* cfg)
+{
+    int error = -1;
+    FILE *fp = fopen(filename, "r");
+    if (!fp) {
+        return -1; // File could not be opened
+    }
+
+    // Move to the end of the file to get its size
```
Comment (MichaelBrim) [inline configurator/configurator.c]: `fseek(END)+ftell()+fseek(BEGIN)` is way more complicated than just calling `stat()` to get the file size

File: configurator/configurator.c [hunk 3]
```diff
@@ -583,53 +585,183 @@ int inih_config_handler(void* user,
...
+void json_config_handler(enum NanoJSONCError error, const char *const key, const char *const value,
+                        const char *const section, void *object)
 {
-    int rc, inih_rc;
-    char errmsg[PREFIX_CFG_MAX_MSG];
+    if (error != NO_ERROR)
+    {
+        return;
+    }
+    char* simple_section = strdup(section);
```
Comment (MichaelBrim) [inline configurator/configurator.c]: all these string dups are being leaked

File: configurator/configurator.c [hunk 4]
```diff
@@ -583,53 +585,183 @@ int inih_config_handler(void* user,
...
 
-    if( NULL == cfg )
-        return EINVAL;
-    
-    if( NULL == file )
-        return EINVAL;
+    if(simple_section)
```
Comment (MichaelBrim) [inline configurator/configurator.c:598]: I understand why this is sufficient, but I would prefer if there was also a NULL check on `section` that precedes the dup

**Review by MichaelBrim — CHANGES_REQUESTED** [review]
Overall, looks good. See specific comments for things to address. When everything is resolved, please squash into one commit.

File: configurator/configurator.c (hunk 2 above)
Comment (SamHerts) [reply configurator/configurator.c]: Ah, didn't realize you already had sys/stat included.

**Comment by SamHerts** [comment]
Changes made and commits squashed.

File: configurator/configurator.c [hunk 5]
```diff
@@ -583,53 +585,181 @@ int inih_config_handler(void* user,
...
+    char errmsg[PREFIX_CFG_MAX_MSG];
+    switch( errcode ) {
+        case 0:
+            rc = 0;
         break;
-    case -1:
-        snprintf(errmsg, sizeof(errmsg),
-                 "failed to open config file %s", 
-                 file);
+        case -1:
```
Comment (MichaelBrim) [inline configurator/configurator.c]: something weird going on with the switch case indents, here and below in `json_error_handler`

Comment (SamHerts) [reply configurator/configurator.c]: IDE must have messed with the indents

**Review by MichaelBrim — APPROVED** [review]
Looks good. I'll merge soon

**Pull request merged**